File: backend/src/main/java/com/smartestate/entity/Complaint.java
package com.smartestate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.smartestate.constants.ComplaintConstants;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("complaints")
public class Complaint {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String category;
    private String title;
    private String content;
    private String status = ComplaintConstants.PENDING;
    private String reply;
    private Long replierId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private User user;
    @TableField(exist = false)
    private User replier;
}

File: frontend/src/types/complaint.ts
import type { ComplaintCategory, ComplaintStatus } from '../constants/complaint';
import type { User } from './user';

export interface Complaint {
  id: number;
  userId: number;
  category: ComplaintCategory;
  title: string;
  content: string;
  status: ComplaintStatus;
  reply?: string;
  replierId?: number;
  createdAt: string;
  updatedAt: string;
  user?: User;
  replier?: User;
}

export interface ComplaintPayload {
  category: ComplaintCategory;
  title: string;
  content: string;
}
